neuralprophet/df_utils.py

```diff
@@ -36,31 +36,32 @@ def prep_or_copy_df(df):
             whether the ID col was present
         bool
             wheter it is a single time series
+        list
+            list of IDs
     """
-    received_ID_col = False
-    received_single_time_series = True
-    if isinstance(df, pd.DataFrame):
-        new_df = df.copy(deep=True)
-        if "ID" in df.columns:
-            received_ID_col = True
-            log.debug("Received df with ID col")
-            if len(new_df["ID"].unique()) > 1:
-                log.debug("Received df with many time series")
-                received_single_time_series = False
-            else:
-                log.debug("Received df with single time series")
-        else:
-            new_df["ID"] = "__df__"
-            log.debug("Received df with single time series")
-    elif df is None:
-        raise ValueError("df is None")
-    else:
-        raise ValueError("Please, insert valid df type (pd.DataFrame)")
+    if not isinstance(df, pd.DataFrame):
+        raise ValueError("Provided DataFrame (df) must be of pd.DataFrame type.")
+
+    # Create a copy of the dataframe
+    df_copy = df.copy(deep=True)
+
+    df_has_id_column = "ID" in df_copy.columns
+
+    # If there is no ID column, then add one with a single value
+    if not df_has_id_column:
+        log.debug("Provided DataFrame (df) contains a single time series.")
```

Comment:
I am wondering if this debug info is necessary or if we could leave it out. Just a detailed thought, however, feel free to decide based on your preferences.

Reply:
We have a lot of logging information which I'm not sure how helpful it actually is, see #982 - as long as it is debug I would keep it for now (meaning it does not show by default anyhow)